### ISSUE TITLE
suggestion: Use ref

### ISSUE THREAD

**Issue by mikey0000** [danomatic/react-pdf-html#5]
Is there any issue with using ref to grab elements and css directly from the node
save the step of rendering the dom to string and back again and be able to grab all styles that exist both inline and in sheet.

**Comment by danomatic** [comment]
In my initial use case I was using HTML for some of my content server side (user editable markdown -> HTML -> PDF). It sounds like you are exploring rendering DOM elements from a browser using their computed styles. That would be great to explore but I just haven't looked into it.

**Comment by mikey0000** [comment]
I may explore it for my use case so I'll PR the changes if it goes ahead and yes you are right I'm rendering from the browser.

**Comment by mikey0000** [comment]
https://github.com/mikey0000/react-pdf-html/tree/feature/use-ref

work in progress. By no means perfect. I'll be working on it for a few days more at least

**Comment by danomatic** [comment]
> https://github.com/mikey0000/react-pdf-html/tree/feature/use-ref
> 
> work in progress. By no means perfect. I'll be working on it for a few days more at least

@mikey0000 awesome. May I suggest a few separate PRs? I'm noticing that you are dealing with multiple things at once in that branch (support for additional tags, further image support, etc)

**Comment by mikey0000** [comment]
Yeah I'll have to spend some time to split it out into usable PRs, just needed to work on getting it working first. There are some hurdles with images but I'll PR svg support(when it's completed which shouldn't be hard) and the additional tags that were missed (section and span)

3/09/2021 9:24:56 am Dan Blaisdell ***@***.***>:
